sycl/test-e2e/AmdNvidiaJIT/kernel_and_bundle.cpp

```diff
@@ -1,5 +1,6 @@
 // UNSUPPORTED: windows
 // REQUIRES: cuda || hip
+// REQUIRES: unsplit-mode
```

Comment:
Will address this comment last to leave open to bike-shedding. Would something like `unsplit-test-mode` work? or (`unsplit-testing`), should the other features (`build-mode`, `run-mode`) follow suit? or are they clear enough.

I don't think i'd want to put the word "split" in this particular feature, since it is exactly what it isn't.

Reply:
yeah of course i dont want to start bikeshedding, `REQUIRES: unsplit-test-mode` is fine with me

Reply:
Won't it be clearer to have something like `only-build-test`, `only-run-test`, `build-and-run-test` instead of `build-mode`, `run-mode`, and `unsplit-test-mode`?

Reply:
> Won't it be clearer to have something like `only-build-test`, `only-run-test`, `build-and-run-test` instead of `build-mode`, `run-mode`, and `unsplit-test-mode`?

The `run-mode` feature is useful because it allows us to have common behaviour between the run-only and unsplit modes, if we didnt have it we would have to mark tests that xfail at run-time as both XFAIL on `only-run-test` and `build-and-run-test`.
 
The `build-mode` feature I could not think of a valid use for, since if we have an xfail at build time we would always also want to mark it as an xfail at run-time since we wouldn't have an executable, also a test in the e2e folder that uses `REQUIRES: build-mode` seems a bit suspicious.

I added something like `build-only`/`run-only`/`unsplit` as a variable within the python scripts, but not as a feature since their usefulness in that context is unclear to me. Rather I'm using them to clean up if statements in the python scripts where we want to check what split-mode we are in.